src/style-installer.js

```diff
@@ -283,6 +270,25 @@ export function makeBodyVisible(doc, opt_waitForServices) {
   }
 }
 
+/**
+ * Set the document's body opacity to 1. Called in error cases.
+ * @param {!Document} doc The document who's body we should make visible.
+ */
+export function makeBodyVisibleRecovery(doc) {
+  dev().assert(doc.defaultView, 'Passed in document must have a defaultView');
+  const win = /** @type {!Window} */ (doc.defaultView);
+  if (win[bodyVisibleSentinel]) {
```

Comment:
You've only moved the nasty bug 😉to down here.

Reply:
That's right. But this should have mostly removed the impact of the Chrome bug. Meantime, the code is also much cleaner this way.

Reply:
> But this should have mostly removed the impact of the Chrome bug. 

Only if there's not an error. If there is, then it'll still hit the 8s delay.

> Meantime, the code is also much cleaner this way.

Agreed, I'm happy with the overall refactor.